### PR TITLE
Remove AP_InertialNav methods

### PR DESCRIPTION
**Pull request by peterbarker**
These were just passing through to EKF anyway.

Make a whole bunch of methods that can fail if we don't have location return bool rather than float.

### CODE REVIEW

File: ArduCopter/inertia.cpp [hunk 1]
```diff
@@ -21,7 +21,12 @@ void Copter::read_inertia()
         current_loc.alt = inertial_nav.get_altitude();
     } else {
         // with inertial nav we can update the altitude and climb rate at 50hz
-        current_loc.alt = pv_alt_above_home(inertial_nav.get_altitude());
+        float alt;
+        if (pv_alt_above_home(inertial_nav.get_altitude(), alt)) {
+            current_loc.alt = alt;
+        } else {
+            current_loc.alt = 0.0f; // retain compatability with inertial_nav
```
Comment (OXINARF) [inline ArduCopter/inertia.cpp]: Shouldn't this be `inertial_nav.get_altitude()`? Also, if `pv_alt_above_home` checks that home is set (as I suggested in previous comment), then we don't need the if above.

File: ArduCopter/mode_auto.cpp [hunk 2]
```diff
@@ -1994,8 +1999,11 @@ void Copter::set_auto_yaw_roi(const Location &roi_location)
 #if MOUNT == ENABLED
         // check if mount type requires us to rotate the quad
         if(!camera_mount.has_pan_control()) {
-            roi_WP = pv_location_to_vector(roi_location);
-            set_auto_yaw_mode(AUTO_YAW_ROI);
+            if (pv_location_to_vector(roi_location, roi_WP)) {
+                set_auto_yaw_mode(AUTO_YAW_ROI);
+            } else {
+                set_auto_yaw_mode(AUTO_YAW_LOOK_AT_HEADING);
```
Comment (OXINARF) [inline ArduCopter/mode_auto.cpp]: Humm, shouldn't we do HOLD instead?

File: ArduCopter/mode_auto.cpp [hunk 3]
```diff
@@ -2008,8 +2016,11 @@ void Copter::set_auto_yaw_roi(const Location &roi_location)
         //      4: point at a target given a target id (can't be implemented)
 #else
         // if we have no camera mount aim the quad at the location
-        roi_WP = pv_location_to_vector(roi_location);
-        set_auto_yaw_mode(AUTO_YAW_ROI);
+        if (pv_location_to_vector(roi_location, roi_WP)) {
+            set_auto_yaw_mode(AUTO_YAW_ROI);
+        } else {
+            set_auto_yaw_mode(AUTO_YAW_LOOK_AT_HEADING);
```
Comment (OXINARF) [inline ArduCopter/mode_auto.cpp]: Same.

File: ArduCopter/commands.cpp [hunk 4]
```diff
@@ -21,34 +21,38 @@ void Copter::update_home_from_EKF()
...
+    }
+    Location ekf_origin;
+    if (!ahrs.get_origin(ekf_origin)) {
+        return;
+    }
+    temp_loc.alt = ekf_origin.alt;
+    if (!set_home(temp_loc, false)) {
+        return;
+    }
         // we have successfully set AHRS home, set it for SmartRTL
```
Comment (OXINARF) [inline ArduCopter/commands.cpp]: Re-indent the comment please. And moving it to inside the macro is probably a good idea.

File: ArduCopter/commands.cpp [hunk 5]
```diff
@@ -21,34 +21,38 @@ void Copter::update_home_from_EKF()
...
-            return false;
-        }
-        // we have successfully set AHRS home, set it for SmartRTL
+    if (!ahrs.get_location(temp_loc)) {
+        return false;
+    }
+    if (!set_home(temp_loc, lock)) {
+        return false;
+    }
+    // we have successfully set AHRS home, set it for SmartRTL
```
Comment (OXINARF) [inline ArduCopter/commands.cpp]: Same thing about putting it inside macro.

File: ArduCopter/commands.cpp [hunk 6]
```diff
@@ -21,34 +21,38 @@ void Copter::update_home_from_EKF()
...
-        temp_loc.alt = ekf_origin.alt;
-        if (!set_home(temp_loc, false)) {
-            return;
-        }
+    if (!ahrs.get_location(temp_loc)) {
+        return;
+    }
+    Location ekf_origin;
+    if (!ahrs.get_origin(ekf_origin)) {
+        return;
```
Comment (OXINARF) [inline ArduCopter/commands.cpp]: Humm I wonder if we should really return here? I think that setting home to current altitude is still better than not setting it (I think this is impossible to happen by the way, if we have location we also have origin, but we should definitely check it).

**Comment by khancyr** [comment]
this is normally already addressed on https://github.com/ArduPilot/ardupilot/pull/7345 and on all vehicles !

**Comment by OXINARF** [comment]
@khancyr I remembered your PR right away when I saw this. As I wrote there, I think some stuff can be cherry-picked to easy transition. Once the diff is smaller, it will make it easier to force people into looking at it and deciding what we want to do.

**Comment by peterbarker** [comment]
On Wed, 4 Apr 2018, Francisco Ferreira wrote:

> AllVehicles but only Copter commits? 😛

I missed the "WIP" label...

> Besides inline comments, as I've said before, I'd prefer we didn't make
> several separate commits removing methods from AP_InertialNav - our

Any particular reason?  If you want it gone (and who doesn't?), this is a 
way that will work, and can be done in a way that won't bitrot.

> objective is removing it entirely and doing it in parts in vehicles is good,
> but seems unnecessary for the library side.

It makes it clearer (and less intimidating) what's going on if we remove 
things as we go IMO.

> In ArduCopter/position_vector.cpp:
> 
> >  {
> -    const struct Location &origin = inertial_nav.get_origin();
> -    return alt_above_home_cm + (ahrs.get_home().alt - origin.alt);
> +    struct Location origin;
> +    if (!ahrs.get_origin(origin)) {
> +        return false;
> +    }
> +    ret = alt_above_home_cm + (ahrs.get_home().alt - origin.alt);
> 
> Shouldn't we check home is set?

Yep!

> Same.

Yep!

>          current_loc.alt = inertial_nav.get_altitude();
>      } else {
>          // with inertial nav we can update the altitude and climb rate at 5
> 0hz
> -        current_loc.alt = pv_alt_above_home(inertial_nav.get_altitude());
> +        float alt;
> +        if (pv_alt_above_home(inertial_nav.get_altitude(), alt)) {
> +            current_loc.alt = alt;
> +        } else {
> +            current_loc.alt = 0.0f; // retain compatability with inertial_n
> av
> 
> Shouldn't this be inertial_nav.get_altitude()? Also, if pv_alt_above_home

You mean in the comment?

> checks that home is set (as I suggested in previous comment), then we don't
> need the if above.

Yes, there are several places where we do additional checks that can be 
removed.  I'm particularly interested in why we ask what the filter status 
is when we can just ask the AHRS for the information...

> +                set_auto_yaw_mode(AUTO_YAW_LOOK_AT_HEADING);
> 
> Humm, shouldn't we do HOLD instead?

That works.  I'm just really worried this is going to conflict horribly 
with the autoyaw patch.

> +    Location ekf_origin;
> +    if (!ahrs.get_origin(ekf_origin)) {
> +        return;
> +    }
> +    temp_loc.alt = ekf_origin.alt;
> +    if (!set_home(temp_loc, false)) {
> +        return;
> +    }
>          // we have successfully set AHRS home, set it for SmartRTL
> 
> Re-indent the comment please. And moving it to inside the macro is probably
> a good idea.

Pretty sure we can do even better than that; I think now that ahrs keeps 
track of home_is_set SmartRTL can just ask it directly...

> -        const struct Location &ekf_origin = inertial_nav.get_origin();
> -        temp_loc.alt = ekf_origin.alt;
> -        if (!set_home(temp_loc, false)) {
> -            return;
> -        }
> +    if (!ahrs.get_location(temp_loc)) {
> +        return;
> +    }
> +    Location ekf_origin;
> +    if (!ahrs.get_origin(ekf_origin)) {
> +        return;
> 
> Humm I wonder if we should really return here? I think that setting home to
> current altitude is still better than not setting it (I think this is
> impossible to happen by the way, if we have location we also have origin,
> but we should definitely check it).

Yeah, I was also wondering about that...

**Comment by amilcarlucas** [comment]
This is a nice one, can we add this to the dev call ?

**Comment by khancyr** [comment]
as remember for later, pv* are the same as the one in ahrs

**Comment by peterbarker** [comment]
@khancyr Which ones are duplicated in `AP_AHRS`?

**Comment by peterbarker** [comment]
Will be faster to redo this than resolve the conflicts.

**Pull request closed without merging**